src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala

```diff
@@ -731,4 +756,18 @@ class GroupReadsByUmi
         r1[String](this.rawTag)
     }
   }
+
+  /** Sets the duplicate flags on all reads within all templates.  */
+  private def setDuplicateFlags(group: Seq[Template]): Unit = {
+    val nonDuplicateTemplate = group.minBy { template =>
+      template.primaryReads.sumBy { r =>
+        DuplicateScoringStrategy.computeDuplicateScore(r.asSam, ScoringStrategy.SUM_OF_BASE_QUALITIES)
```

Comment:
As a compromise, consider setting the ScoringStrategy as a constructor parameter that's not exposed on the command line?

Reply:
I really don't want to expose this anywhere - I want the freedom to change this if/when we decide using HTSJDK's duplicate scoring stuff was a bad idea in the future, without breaking any APIs.

Reply:
A few things in here - 
1.) hard-coding this seems like a really odd choice, we could wrap it if you're concerned about leaking API.
2.) you don't have a tie break, which is why mapQ was included in my implementation. 

this is more tidy logic, nice.

Reply:
Just as a point re duplicate scoring strategies - the core strategies for choosing the representative read, much like UMI clustering algorithms are fairly established in literature & while having others seems possible, not supporting established ones, seems like a limitation which would be self imposed, I'm happy to wrap these enums if that's a need from the projects perspective so that the HTSJDK api isn't leaking.

Reply:
Lol, this did make me laugh a little.  I would agree if deep thought had gone into the original duplicate scoring strategies ... but as the author of that original code (in Picard that moved into HTSJDK) I can tell you that it was me making it up on the fly and coming up with something vaguely sensible!

Over time I tend to think that the duplicate marking stuff in Picard has gotten _far_ too complicated.

I'd also argue that there's little to no value tie-breaking on mapq.  If the reads are considered duplicates of each other and have either a) the same sum of base qualities > Q15, or b) the same mapped reference length, there's a very high probability they also have the same sum of mapqs.

Reply:
I don't necessarily mean the algorithm was fully flushed out, i mean it's usage is well established in literature. 
Regardless - I'd be just as happy to implement a version of UMItools which is just as or more vetted. https://umi-tools.readthedocs.io/en/latest/reference/dedup.html#selecting-the-representative-read

I'd tend to argue that baseQ isn't a very important metric at this stage anyway mapping quality is far more important, since it has already taken the base q into consideration when being aligned. 

Agree most of picards stuff is too complex - which is why i did the simplest thing i could think of.

Reply:
I'm happy to use something different when picking the representative, though would also like in this first version to just have one way of doing it, and not a user option.

My intuition though is that if you used mapq (summed across primary reads?), then in the vast majority of cases it just becomes a random selection.  The most prevelant short read aligners (bwa, star, minimap2) don't use base quality in computing mapping quality, so reads that are duplicates of one another are likely to have identical mapping qualities _except_ in regions that are hard to map, where mapping a few more or fewer bases is likely to change the confidence.

How about mapq + total number of mapped query bases?  Or something along those lines?

Reply:
Alternatively @JoeVieira I'd be happy to be proved wrong - if you want to take a look at some real data and report how frequently there are differing maqps within a tag family.

Reply:
One options seems fine. I think it's perfectly fine that that is sum of base q, nothing about picards implementation of these scoring strategies is complex, the usages deep in the code get to be rather complex, but this simple score calculator just seems to be good code reuse to me. 

I'm happy to look at how map q changes of note my testing is with bowtie2 mostly - so that certainly is impacting my perspective here.

I'll stop pushing here & just test a few more things, give some perspective with data.

Reply:
Thanks @JoeVieira ... please let me know when you've had a chance to look at your data.

Reply:
@tfenne it'll be a little bit before I'm able to dig back into that data. But will in the next week or so.

Reply:
I think it would be sensible to choose the read with highest mapq and breaking ties randomly.

@JoeVieira did you have any suggestions after reviewing some data?

Reply:
Sorry all, I have no been able to review data at scale yet. It's on my list for next week i promise.